packages/ui/src/pages/Home/Home.tsx

```diff
@@ -93,30 +107,27 @@ const Home = ({ className }: HomeProps) => {
         </h3>
       </CenterStyled>
     )
+  }
 
-  if (!api || isAccountLoading) {
+  if (!api) {
     return (
-      <Box
-        className={className}
-        sx={{
-          display: 'flex',
-          flexDirection: 'column',
-          alignItems: 'center',
-          '&:first-of-type': {
-            marginBottom: '1rem'
-          }
-        }}
-        data-cy="loader-accounts-rpc-connection"
-      >
+      <LoaderBoxStyled data-cy="loader-rpc-connection">
+        <CircularProgress />
+        {`Connecting to the node at ${selectedNetworkInfo?.rpcUrl}`}
+      </LoaderBoxStyled>
+    )
+  }
+
+  if (isAccountLoading) {
+    return (
+      <LoaderBoxStyled data-cy="loader-accounts-connection">
         <CircularProgress />
-        {isAccountLoading
-          ? 'Loading accounts...'
-          : `Connecting to the node at ${selectedNetworkInfo?.rpcUrl}`}
-      </Box>
+        Loading accounts...
+      </LoaderBoxStyled>
     )
   }
 
-  if (isLoading) {
+  if (isLoadingMultisigs) {
```

Comment:
What about to move the loading and error handling in the separate file:
```jsx
import React from 'react';
import { CircularProgress } from '@mui/material';
import { LoaderBoxStyled, MessageWrapper, ErrorMessageStyled } from './YourStyledComponents'; // Update with your actual imports

const useLoadingAndError = ({ api, isAccountLoading, isLoadingMultisigs, multisigQueryError, multiProxyList, showNewMultisigAlert, selectedNetworkInfo }) => {
  if (!api) {
    return (
      <LoaderBoxStyled data-cy="loader-rpc-connection">
        <CircularProgress />
        {`Connecting to the node at ${selectedNetworkInfo?.rpcUrl}`}
      </LoaderBoxStyled>
    );
  }

  if (isAccountLoading) {
    return (
      <LoaderBoxStyled data-cy="loader-accounts-connection">
        <CircularProgress />
        Loading accounts...
      </LoaderBoxStyled>
    );
  }

  if (isLoadingMultisigs) {
    return (
      <MessageWrapper>
        <CircularProgress />
        <div>Loading your multisigs...</div>
      </MessageWrapper>
    );
  }

  if (multisigQueryError) {
    return (
      <MessageWrapper>
        <ErrorMessageStyled>
          <ErrorOutlineIcon size={64} />
          <div>An error occurred.</div>
        </ErrorMessageStyled>
      </MessageWrapper>
    );
  }

  if (multiProxyList.length === 0) {
    return (
      <MessageWrapper>
        {showNewMultisigAlert ? <SuccessCreation /> : <ConnectOrWatch />}
      </MessageWrapper>
    );
  }

  return null; 
};

export default useLoadingAndError;
```

Home page
```jsx
  const loadingAndErrorComponent = useLoadingAndError({
    api,
    isAccountLoading,
    isLoadingMultisigs,
    multisigQueryError,
    multiProxyList,
    showNewMultisigAlert,
    selectedNetworkInfo,
  });

  if (loadingAndErrorComponent) {
    return loadingAndErrorComponent;
  }
  ```

Reply:
Sure, that'd help make the Home easier to read, I'll do it in a normal component though, rather than a hook.

Reply:
I pushed an update separating `DisplayError` and `DisplayLoader`, and keeping the rest as is for now.